pkg/authorization/static/static.go

```diff
@@ -38,8 +38,9 @@ type StaticAuthorizationConfig struct {
 }
 
 type UserConfig struct {
-	Name   string   `json:"name,omitempty"`
-	Groups []string `json:"groups,omitempty"`
+	Name     string   `json:"name,omitempty"`
+	Groups   []string `json:"groups,omitempty"`
+	GroupSet map[string]struct{}
```

Comment:
Use [k8s.io/utils/set](https://github.com/kubernetes/utils/blob/24370beab75816ab1d2b33fba9e288ee667530dd/set/set.go#L28-L35):

```suggestion
	GroupSet set.Set[string]
```